### DISCUSSION TITLE
Batch processing - is it possible?

### DISCUSSION THREAD

**Discussion by poetsmeniet** [OP-TED/ted-xml-data-converter#24] [Q&A] [answered]
Converting file by file requires the jvm to be spawned or each conversion. Is it possible to speed this up with batch processing?

**Comment by pdonohoe** [comment, marked as answer]
That's an interesting question, @poetsmeniet. I develop the XSLT by running the XSLT conversion in Saxon 10HE from the unix command line, and it only allows conversion of one XML file at a time: https://www.saxonica.com/html/documentation10/using-xsl/commandline/index.html. I imagine it would be possible to do multiple conversions by using the JAR file from within a Java or C++ program.
I have used a method to use the same Saxon 10HE to run XQuery scripts on more than one XML file at a time, but this requires reading the XML files from within the XQuery script. It is possible to do something similar with XSLT, using the fn:doc() function to read XML files, and the <xsl:result-document> instruction to write files, but I have not found a sufficient use-case for me to develop this.
You could write a wrapper XSLT file which includes or imports the ted-to-eforms.xslt stylesheet, and calls the starting template once per file, but you would also need to make some changes to the ted-to-eforms.xslt stylesheet.

**Reply by poetsmeniet** [reply]
Hi Paul,

As I am not that familiar with XSLT I found another solution that is specifically made for this use case. It's not lightning fast, but for our purposes is sufficient.

It's an open source project:
https://github.com/facebookarchive/nailgun
Nailgun is a client, protocol, and server for running Java programs from the command line without incurring the JVM startup overhead.

Hope this helps anyone with the same requirement.

**Reply by pdonohoe** [reply]
Hi @poetsmeniet , I'm glad you found an acceptable solution. I will look at nailgun for my own projects.
